pyne/alara.py

```diff
@@ -70,18 +95,35 @@ def mesh_to_fluxin(flux_mesh, flux_tag, fluxin="fluxin.out",
         direction = -1
 
     output = ""
-    for i, mat, ve in flux_mesh:
-        # print flux data to file
-        count = 0
-        flux_data = np.atleast_1d(tag_flux[ve])
-        for i in range(start, stop, direction):
-            output += "{:.6E} ".format(flux_data[i])
-            # fluxin formatting: create a new line after every 6th entry
-            count += 1
-            if count % 6 == 0:
-                output += "\n"
-
-        output += "\n\n"
+    if not sub_voxel:
+        for i, mat, ve in flux_mesh:
+            # print flux data to file
+            count = 0
+            flux_data = np.atleast_1d(tag_flux[ve])
+            for i in range(start, stop, direction):
+                output += "{:.6E} ".format(flux_data[i])
+                # fluxin formatting: create a new line after every 6th entry
+                count += 1
+                if count % 6 == 0:
+                    output += "\n"
+
+            output += "\n\n"
+
+    else:
+        for row in cell_fracs:
+            for i, mat, ve in flux_mesh:
+                if i == row['idx'] and len(cell_mats[row['cell']].comp) != 0:
+                    count = 0
+                    flux_data = np.atleast_1d(tag_flux[ve])
+                    for i in range(start, stop, direction):
+                        output += "{:.6E} ".format(flux_data[i])
+                        # fluxin formatting: create a new line
+                        # after every 6th entry
+                        count += 1
+                        if count % 6 == 0:
+                            output += "\n"
+
+                    output += "\n\n"
```

Comment:
Since the body of these two clause is mostly the same, can you maybe please provide a private method (ie a method that begins with an underscore `_`) or similar that performs the work.  This function would then just be called with different arguments.  There is a lot of code duplication here that could be removed :) Thanks!

Reply:
Thanks for your suggestion @scopatz . I put these two clause into a private method `_output_flux()` to reduce duplication.